javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -272,72 +272,271 @@ public static boolean isSelfAccess(final ExpressionTree tree) {
+  // This section of the file groups methods by their receiver type; that is, it puts all
+  // `elementFrom*(FooTree)` methods together.
+
+  /**
+   * Returns the type element corresponding to the given class declaration.
+   *
+   * @param tree class declaration
+   * @return the element for the given class
+   */
+  public static @Nullable TypeElement elementFromDeclaration(ClassTree tree) {
+    TypeElement result = (TypeElement) TreeInfo.symbolFor((JCTree) tree);
+    return result;
+  }
+
   /**
-   * Gets the {@link Element} for the given Tree API node.
+   * Returns the type element corresponding to the given class declaration.
+   *
+   * <p>The TypeElement may be null for an anonymous class.
    *
    * @param tree the {@link Tree} node to get the symbol for
-   * @throws IllegalArgumentException if {@code tree} is null or is not a valid javac-internal tree
-   *     (JCTree)
    * @return the {@link Symbol} for the given tree, or null if one could not be found
+   * @deprecated use elementFromDeclaration
    */
+  @Deprecated // not for removal; retain to prevent calls to this overload
   @Pure
-  public static TypeElement elementFromTree(ClassTree tree) {
-    return (TypeElement) TreeInfo.symbolFor((JCTree) tree);
+  public static @Nullable TypeElement elementFromTree(ClassTree tree) {
+    return elementFromDeclaration(tree);
   }
 
   /**
-   * Gets the {@link Element} for the given Tree API node.
+   * Returns the type element corresponding to the given class declaration.
+   *
+   * @param tree the {@link Tree} node to get the symbol for
+   * @return the {@link Symbol} for the given tree, or null if one could not be found
+   * @deprecated use elementFromDeclaration
+   */
+  @Deprecated // not for removal; retain to prevent calls to this overload
+  @Pure
+  public static @Nullable TypeElement elementFromUse(ClassTree tree) {
+    return elementFromDeclaration(tree);
+  }
+
+  /**
+   * Returns the element corresponding to the given use.
+   *
+   * @param tree the tree corresponding to a use of an element
+   * @return the element for the corresponding declaration, {@code null} otherwise
+   */
+  @Pure
+  @Deprecated // not for removal; retain to prevent calls to this overload
+  public static @Nullable Element elementFromDeclaration(ExpressionTree tree) {
+    return TreeUtils.elementFromUse(tree);
+  }
+
+  /**
+   * Returns the element corresponding to the given use.
+   *
+   * @param tree the tree corresponding to a use of an element
+   * @return the element for the corresponding declaration, {@code null} otherwise
+   * @deprecated use elementFromUse
+   */
+  @Deprecated // not for removal; retain to prevent calls to this overload
+  @Pure
+  public static @Nullable Element elementFromTree(ExpressionTree tree) {
+    return TreeUtils.elementFromUse(tree);
+  }
+
+  /**
+   * Returns the element corresponding to the given use.
+   *
+   * @param tree the tree corresponding to a use of an element
```

Comment:
Not all ExpressionTrees are uses of an element.  For example, BinaryTrees are ExpressionTrees, but they are not uses of an element.  I would deprecate this method and use  `Element elementFromTree(ExpressionTree tree)` or `Element elementFromTree(Tree tree)` instead.

Reply:
Did you intentional not mark this as deprecate or was it an oversight?  The Javadoc here is still incorrect. `tree` is an expression which may or may not be the use of an element.